### ISSUE TITLE
Could not invoke BlinkCardAndroid.scanWithCamera

### ISSUE THREAD

**Issue by KevinToh0329** [blinkcard/blinkcard-react-native#9]
Hi, I'm currently facing this issue out of sudden. Not sure what's the issue here. Hope anyone can help me.
currently using @microblink/blinkcard-react-native @ v2.9.0

I tried to console.log the error at the catch. It doesn't appear anything.

```
startScan = async () => {
    let recognizer; 
    recognizer = new BlinkCardReactNative.BlinkCardRecognizer();
    recognizer.extractCvv = false;
    recognizer.extractIban = false;
    recognizer.extractExpiryDate = false;
    recognizer.extractOwner = false;
    recognizer.returnFullDocumentImage = true;

    if (recognizer) {
        try {
            // logger appear here
            const scanningResults = await BlinkCardReactNative.BlinkCard.scanWithCamera(
                new BlinkCardReactNative.BlinkCardOverlaySettings(),
                new BlinkCardReactNative.RecognizerCollection([recognizer]),
                licenseKey
            ).catch((error) => { 
              // no logger appear here.
              throw error;
            });
            // no logger appear after here.
            if (scanningResults.length > 0) {
                this.onResult(scanningResults[0]); // A function to insert into variable
            }
        } catch (error) {
            alert(JSON.stringify(error))
        }
    }
}
```
![Could not invoke BlinkCardAndroid.scanWithCamera](https://github.com/blinkcard/blinkcard-react-native/assets/34225944/445943b6-dc9b-4c0e-8d82-55bdafa30219)

**Comment by mparadina** [comment]
Hi @KevinToh0329 

Just to double-check, are you using the BlinkCard SDK on a simulator or a physical device?
The SDK is not supported for running on simulators.

When does the exception occur? 
When initializing the SDK, during the scanning process, or when you handle the scanned results?

**Comment by KevinToh0329** [comment]
Hi @mparadina ,
Thanks for the quick response.

It's on a Physical Device.

The exception occur when this line of code run.
```
await BlinkCardReactNative.BlinkCard.scanWithCamera(
    new BlinkCardReactNative.BlinkCardOverlaySettings(),
    new BlinkCardReactNative.RecognizerCollection([recognizer]),
    licenseKey
)
```
Which I guess is during initializing the SDK, right before scanning process.. Because the camera didn't appear at all.

**Comment by mparadina** [comment]
@KevinToh0329 thank you for the feedback.

Have you used the BlinkCard SDK before and it was working as expected? 
This code looks correctly implemented. 

Just to double-check, when you obtained the SDK, do you have [Git LFS](https://git-lfs.com/) installed? 
It is required to correctly obtain the BlinkCard framework.

Lastly, is the issue occurring with our sample application? You only need to `git clone` this repo, enter it, and run the [./initReactNativeSampleApp.sh](https://github.com/blinkcard/blinkcard-react-native/blob/main/initReactNativeSampleApp.sh) script via Terminal.

**Comment by KevinToh0329** [comment]
Hi @mparadina ,
The BlinkCard SDK was working fine before. We installed this SDK for years.

I uses `npm i`. because the package already marked in the `package.json`.

```
    "@microblink/blinkcard-react-native": "^2.4.1",
```

and inside `package-lock.json`
```
    "@microblink/blinkcard-react-native": {
      "version": "2.9.0",
      "resolved": "https://registry.npmjs.org/@microblink/blinkcard-react-native/-/blinkcard-react-native-2.9.0.tgz",
      "integrity": "sha512-qSP+LP6XQYepATqNqdMxxRCovB/zGG8xPv61rV2gtpU7jPFKe1b/K1TnVnkCmyEmRIm5htzSbmzgHyG2QxZ8Yw=="
    },
```

After recent deployment to production, we faces the issue where the APP crashes when the user click on the Card OCR.

**Comment by mparadina** [comment]
Hi @KevinToh0329 

We've released [BlinkCard 2.9.1](https://github.com/blinkcard/blinkcard-react-native/releases/tag/v2.9.1) in the meantime. Can you try obtaining it and seeing the issue persists?
You can just run the command:  `npm install --save @microblink/blinkcard-react-native`

Please try this out and let me know if it helped.

**Comment by KevinToh0329** [comment]
Hi @mparadina,

I'm currently facing issue when trying to compile..

Here's the error logs...
```
> Task :@microblink_blinkcard-react-native:compileDebugJavaWithJavac FAILED
../node_modules/@microblink/blinkcard-react-native/src/android/src/main/java/com/microblink/blinkcard/reactnative/MicroblinkModule.java:103: error: local variable promise is accessed from within inner class; needs to be declared final
                promise.reject(throwable);
                ^
../node_modules/@microblink/blinkcard-react-native/src/android/src/main/java/com/microblink/blinkcard/reactnative/MicroblinkModule.java:120: error: local variable backImage is accessed from within inner class; needs to be declared final
                        if (backImage != null) {
                            ^
../node_modules/@microblink/blinkcard-react-native/src/android/src/main/java/com/microblink/blinkcard/reactnative/MicroblinkModule.java:121: error: local variable backImage is accessed from within inner class; needs to be declared final
                            processImage(backImage.getString(PARAM_BACK_IMAGE), mScanResultListenerBackSide);
                                         ^
../node_modules/@microblink/blinkcard-react-native/src/android/src/main/java/com/microblink/blinkcard/reactnative/MicroblinkModule.java:121: error: local variable mScanResultListenerBackSide is accessed from within inner class; needs to be declared final
                            processImage(backImage.getString(PARAM_BACK_IMAGE), mScanResultListenerBackSide);
                                                                                ^
../node_modules/@microblink/blinkcard-react-native/src/android/src/main/java/com/microblink/blinkcard/reactnative/MicroblinkModule.java:125: error: local variable promise is accessed from within inner class; needs to be declared final
                            handleDirectApiError("Could not extract the information from the front side and back side is empty!", promise);
                                                                                                                                  ^
../node_modules/@microblink/blinkcard-react-native/src/android/src/main/java/com/microblink/blinkcard/reactnative/MicroblinkModule.java:135: error: local variable promise is accessed from within inner class; needs to be declared final
                    handleDirectApiError("Could not extract the information with DirectAPI!", promise);
                                                                                              ^
../node_modules/@microblink/blinkcard-react-native/src/android/src/main/java/com/microblink/blinkcard/reactnative/MicroblinkModule.java:140: error: local variable promise is accessed from within inner class; needs to be declared final
                promise.reject(throwable);
                ^
../node_modules/@microblink/blinkcard-react-native/src/android/src/main/java/com/microblink/blinkcard/reactnative/MicroblinkModule.java:197: error: local variable promise is accessed from within inner class; needs to be declared final
                handleDirectApiError("Failed to initialize recognizer with DirectAPI: " + throwable.getMessage(), promise);
                                                                                                                  ^
Note: ../node_modules/@microblink/blinkcard-react-native/src/android/src/main/java/com/microblink/blinkcard/reactnative/MicroblinkModule.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
8 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':@microblink_blinkcard-react-native:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.4.1/userguide/command_line_interface.html#sec:command_line_warnings
```

**Comment by KevinToh0329** [comment]
Hi again,

When I try to install older version. I'm hitting this error. And there's no additional logs to check with it.

![image](https://github.com/blinkcard/blinkcard-react-native/assets/34225944/b57b6d09-0160-4e92-9da1-93c377f2a326)

**Comment by mparadina** [comment]
Hi @KevinToh0329 

When you try installing the sample application with BlinkCard 2.9.1, are you getting the exact same `local variable` exceptions? I tried reproducing the same behavior on my end but with no luck.

Are you maybe able to share a minimal reproduction sample so that we can try to reproduce it on our end?
You can directly reach out to us via mail support@microblink.com

**Comment by mparadina** [comment]
Hi @KevinToh0329 as the inquiry was resolved directly, I am closing the issue.

The issue with the 2.9.1 version was connected to compatibility issues with React-Native. 
The version used in the project is 0.59.0, while the version used to build and test the latest SDK is 0.72.6.

The issue with the 2.6.0 version was related to anonymizing the card information, which was fixed in the 2.9.0 version.
